### DISCUSSION TITLE
How to get streamID in stream callback?

### DISCUSSION THREAD

**Discussion by ami-GS** [microsoft/msquic#2255] [Q&A] [answered]
In stream callback function, like `StreamCallback(HQUIC Stream, void* Context, QUIC_STREAM_EVENT* Event);`.
How do we know the streamID associated with Stream which call the callback?
I found suspicious ID in QUIC_STREAM_EVENT as bellow, but I don't think this is the one.
```C
typedef struct QUIC_STREAM_EVENT {
    QUIC_STREAM_EVENT_TYPE Type;
    union {
        struct {
            QUIC_STATUS Status;
            QUIC_UINT62 ID;    // <---- This?
            BOOLEAN PeerAccepted : 1;
            BOOLEAN RESERVED : 7;
        } START_COMPLETE;
```

**Comment by nibanks** [comment, marked as answer]
There is a GetParam for the stream to query it's ID. The API isn't super efficient, so I'd recommend caching it if you need it often.

**Reply by ami-GS** [reply]
Thanks! will check the Get/SetParam for other use case as well
